### PR TITLE
tests: wait for region removal

### PR DESCRIPTION
**Pull request by BusyJay**
@siddontang @hhkbp2 PTAL

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

File: tests/raftstore/test_stale_peer.rs [hunk 1]
```diff
@@ -152,10 +152,23 @@ fn test_stale_peer_without_data<T: Simulator>(cluster: &mut Cluster<T>) {
 
     pd_client.must_remove_peer(new_region_id, new_peer(3, 4));
 
-    // Wait for max_leader_missing_duration to time out.
-    thread::sleep(cluster.cfg.raft_store.max_leader_missing_duration);
-    // Sleep one more second to make sure there is enough time for the peer to be destroyed.
-    thread::sleep(Duration::from_secs(1));
+    let timer = Instant::now();
```
Comment (hhkbp2) [inline tests/raftstore/test_stale_peer.rs]: Please modify `test_stale_peer_out_of_region` as this test case to keep them consistent.

**Comment by hhkbp2** [comment]
Rest LGTM

**Comment by hhkbp2** [comment]
LGTM

**Pull request merged**